flixel/input/gamepad/ButtonID.hx

```diff
@@ -0,0 +1,57 @@
+package flixel.input.gamepad;
+
+import flixel.system.macros.FlxMacroUtil;
+
+/**
+ * Maps enum values and strings to integer keycodes.
+ */
+@:enum
+abstract ButtonID(Int) from Int to Int
+{
+	public static var fromStringMap(default, null):Map<String, ButtonID>
+		= FlxMacroUtil.buildMap("flixel.input.gamepad.ButtonID");
+		
+	public static var toStringMap(default, null):Map<ButtonID, String>
+		= FlxMacroUtil.buildMap("flixel.input.gamepad.ButtonID", true);
+	
+	// Button Indices
+	var ANY               = -2;		
+	var NONE              = -1;		
+	var A                 =  0;		//BOTTOM face button
+	var B                 =  1;		//RIGHT  face button
+	var X                 =  2;		//LEFT   face button
+	var Y                 =  3;		//TOP    face button
+	var LEFT_SHOULDER     =  4;		//left "bumper"
+	var RIGHT_SHOULDER    =  5;		//right "bumper"
+	var BACK              =  6;		//a.k.a "select", left center button
+	var START             =  7;		//right center button
+	var LEFT_STICK_BTN    =  8;		//Click left stick
```

Comment:
Well... so I suppose this is a mark against the name "ButtonID." 

More properly, this is a list of numerically unique identifiers for various different control inputs on the device. You pass in one of these IDs to the general function you want, and then it uses the controllers model to map that to the propery raw hardware ID necessary for the final lookup. Internally it maps these universal ids to buttons in justpressed/pressed/justreleased lookups, but it's also used to fetch the axis for analog trigger lookups, and it is also used to fetch the analogStick object (with both axes) for the analog stick x/y axis lookup.